librosa/core/constantq.py

```diff
@@ -23,23 +26,23 @@
 
 @cache(level=20)
 def cqt(
-    y,
+    y: np.ndarray,
     *,
-    sr=22050,
-    hop_length=512,
-    fmin=None,
-    n_bins=84,
-    bins_per_octave=12,
-    tuning=0.0,
-    filter_scale=1,
-    norm=1,
-    sparsity=0.01,
-    window="hann",
-    scale=True,
-    pad_mode="constant",
-    res_type="soxr_hq",
-    dtype=None,
-):
+    sr: float = 22050,
+    hop_length: int = 512,
+    fmin: Optional[float] = None,
+    n_bins: int = 84,
+    bins_per_octave: int = 12,
+    tuning: Optional[float] = 0.0,
+    filter_scale: float = 1,
+    norm: float = 1,
```

Comment:
I think technically `norm` can be Optional here, as the normalize function supports `None` to directly bypass normalization.  It's not documented or encouraged here though, so I'm not sure if we want to flag it as optional.  (If so, the docstring should also be corrected.)

Reply:
OK, since you're not sure, I'm even less sure - so up to you:)

Reply:
I tested it, and it does indeed work with `norm=None`.  (Or, at least, it computes without throwing an exception!)

I think for consistency, we should allow it as optional.

Reply:
Done